keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -110,21 +123,20 @@ When a workload (app pod, deployment, configs) is moved to another cluster, as l
 
 + _backend bucket_ - any bucket provided by the object store system, completely separate from Kubernetes.
 + _brownfield bucket_ - an existing backend bucket.
-+ _Bucket (Bucket Instance)_ - a cluster-scoped custom resource referenced by a `BucketRequest` and containing connection information and metadata for a backend bucket.
++ _Bucket (Bucket instance)_ - a cluster-scoped custom resource referenced by a `BucketRequest` and containing connection information and metadata for a backend bucket.
 + _BucketAccess (BA)_ - a cluster-scoped custom resource for granting bucket access.
 + _BucketAccessClass (BAC)_ - a cluster-scoped custom resource containing fields defining the provisioner and a ConfigMap reference where policy is defined.
 + _BucketAccessRequest (BAR)_ - a user-namespaced custom resource representing a request for access to an existing bucket.
 + _BucketClass (BC)_ - a cluster-scoped custom resource containing fields defining the provisioner and an immutable parameter set for creating new buckets.
-+ _BucketRequest (BR)_ - a user-namespaced custom resource representing a request for a new backend bucket, or access to an existing bucket.
++ _BucketRequest (BR)_ - a user-namespaced custom resource representing a request for a new backend bucket.
 + _COSI_ - Container _Object_ Store Interface, modeled after CSI.
 + _cosi-node-adapter_ - a pod per node which receives Kubelet gRPC _NodeStageVolume_ and _NodeUnstageVolume_ requests, ensures the target bucket has been provisioned, and notifies the kubelet when the pod can be run.
```

Comment:
We need a mechanism to delay to pod from starting until the bucket is available
- it improves the overall user experience
- It puts all the information in one place for the pod to utilize, instead of expecting the pod to figure out how to collect credentials and bucket information for a COSI provisioned bucket

Reply:
I think the "delay" semantic is dubious - apps already have to handle failures, so this gives them a false sense of stability.

The accumulation of coordinates and credentials is a reasonable answer, but this feels like a LOT of machinery for that.  If I were trying to reduce scope I'd maybe suggest that the BucketAccess contain the name of the secret which COSI will populate with well-known keys, and tell the app to use the existing secrets mechanisms.  Am I still over-reducing?

Reply:
As discussed on slack: I think all the mechanics you need exist in secrets already.

I create a BAR and set `secretName: foo`.  I run my pod with a secret volume for foo.  Kubelet sees that foo doesn't exist and waits until it does.  COSI waits for the bucket, then creates foo.  Kubelet can now run the pod.  My pod gets the bucket info.

You can rip out 50% of this KEP

Reply:
and if not, you can open a followup KEP to add CSI integration :)